tests/integ/modin/test_merge_asof.py

```diff
@@ -259,16 +261,18 @@ def test_merge_asof_negative_non_numeric_on(left_right_native_df_non_numeric_on)
     )
     # pandas raises a ValueError with incompatible merge dtype
     with pytest.raises(
-        ValueError,
-        match="Incompatible merge dtype, dtype\\('O'\\) and dtype\\('O'\\), both sides must have numeric dtype",
+        MergeError,
+        match=re.escape(
+            "Incompatible merge dtype, dtype('O') and dtype('O'), both sides must have numeric dtype"
+        ),
     ):
         native_pd.merge_asof(left_native_df, right_native_df, on="a")
     # Snowpark pandas raises a SnowparkSQLException
     # MATCH_CONDITION clause is invalid: The left and right side expressions must be numeric or timestamp expressions.
     with pytest.raises(
         SnowparkSQLException,
     ):
-        pd.merge_asof(left_snow_df, right_snow_df, on="a")
+        pd.merge_asof(left_snow_df, right_snow_df, on="a").to_pandas()
```

Comment:
@sfc-gh-nkrishna are those test skipped in the merge gate? how com it didn't fail in the merge gate?

Reply:
Yeah I'm not sure about that as well, it shouldn't be skipped....

Reply:
They failed when the SQL optimizer was disabled which only run in github daily.

Reply:
@sfc-gh-azhan do you mean SqlSimpflication?

Reply:
that kind of make sense, the sql simplification tiggers an describing call when extracting ColumnState for select, that could explain it

Reply:
Oh yes